pyoptsparse/pyOpt_optimization.py

```diff
@@ -229,71 +240,11 @@ def addVarGroup(
         if varType not in ["c", "i", "d"]:
             raise Error("Type must be one of 'c' for continuous, 'i' for integer or 'd' for discrete.")
 
-        # ------ Process the value argument
-        value = np.atleast_1d(value).real
-        if len(value) == 1:
-            value = value[0] * np.ones(nVars)
-        elif len(value) == nVars:
-            pass
-        else:
-            raise Error(
-                f"The length of the 'value' argument to addVarGroup is {len(value)}, "
-                + f"but the number of variables in nVars is {nVars}."
-            )
-
-        if lower is None:
-            lower = [None for i in range(nVars)]
-        elif np.isscalar(lower):
-            lower = lower * np.ones(nVars)
-        elif len(lower) == nVars:
-            lower = np.atleast_1d(lower).real
```

Comment:
We throw around `.real` very liberally here, IMO this is bad practice and can mask bugs in the code. It is the user's responsibility to make sure things are passed in with the correct dtype, and we should not just cast things to real here. Note that this is potentially a breaking change for people who have CS bugs.

Reply:
I was wondering the same thing, could not figure out why this was here in the first place. I can see how this could break things if something is not working on the user side. When we make a new release, we should highlight this!